### PR TITLE
Update bazel-build-builder.sh

### LINKED ISSUE #3248: Parameterize hard coded environment variables in bazel-build-builder.sh
**Is your feature request related to a problem? Please describe**:
Currently the bazel-build-builder.sh script has the registry and conditions under which it is run hard-coded.
This is awkward for the development and test of new builds for new architectures. 
Also it just seems wrong to have the target registry and run conditions hard coded like that

**Describe the solution you'd like**:
Suggest we make the current hard-coded environment variables the default values, so we can change them on the fly in test if we wish.  Also, the conditions under which the builder image will be re-built are rather absolute.  Suggest we have a boolean environment variable 'ADHOC' which, if set to true, will bypass the test for whether this is a post-submit job which otherwise prevents the builder image from being built or pushed.

**Describe alternatives you've considered**:
We've been considering restructuring CDI build system to be more like the kubevirt build system, and documenting a single consistent set of environment variables, but that's a much larger project. This fix will at least allow us to make easier progress getting there. 

Since bazel-build-builder.sh also fails if it can't push the image to a repo, it would be nice to separate out the build function from the push function, but that can be a separate issue. 

**Additional context**:
Although this feature request is made in support of upstreaming an s390x port, it is not s390x specific

### PR DESCRIPTION
**Pull request by cfilleke**
- uses DOCKER_PREFIX to set the UNTAGGED_BUILDER_IMAGE 
 - separates out the logic to calculate whether this is a POST_SUBMIT job from the test to execute builder build or not
 - tests to see if ADHOC_BUILDER is set so we can more easily build test builder images in development by setting this env var
 - see https://github.com/kubevirt/containerized-data-importer/issues/3248

**What this PR does / why we need it**:

When porting CDI to other platforms or debugging the builder image, we need to do ADHOC_BUILDER images for test. By adding a test to see if this environment variable is set in addition to testing whether this is a POST_SUBMIT build, we can more easily build test CDI builder images. Using the existing DOCKER_PREFIX as set in the config.sh script, the registry and repo is set more consistently with the build system. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3248 

**Special notes for your reviewer**:

This is an incremental step towards a more consistent build system. 

**Release note**:

```release-note
Fixes #3248 
```

### CODE REVIEW

File: hack/build/bazel-build-builder.sh [hunk 1]
```diff
@@ -28,15 +28,29 @@ if [ "${CDI_CONTAINER_BUILDCMD}" = "buildah" ]; then
...
+
+# The other circumstance in which we need to build the builder image is
+# in the course of test and development of the builder image itself. 
+# we'll signal this circumstance by setting the env variable ADHOC_BUILDER
+#
+# also instead of hard-coding the UNTAGGED_BUILDER_IMAGE, we're going
+# to use DOCKER_PREFIX as it is set in config.sh and used elsewhere in 
+# the build system, to introduce a little more consistency  
+
+if ((POST_SUBMIT)) || [ x"${ADHOC_BUILDER}" != "x" ] ; then
```
Comment (awels) [inline hack/build/bazel-build-builder.sh:47]: Why not use `-eq` and `-z` or `-n` so something like
```bash
git diff-index --quiet HEAD~1 hack/build/docker

if [[ $? -eq 0 ]] || [[ -n ${ADHOC_BUILDER} ]]; then
...
fi
```

Comment (mhenriks) [reply hack/build/bazel-build-builder.sh:47]: @awels if you feel strongly about this feel free to hold the PR

Comment (awels) [reply hack/build/bazel-build-builder.sh:47]: I don't feel super strongly about it. It is not like this is run a ton of times and requires optimization. We are creating subshells with the `$((1-$?))` where a simple number comparison will do.

**Comment by cfilleke** [comment]
I wonder why pull-containerized-data-importer-e22-upg keeps failing. First it was a timeout, but now we see

```
 + make cluster-sync
./cluster-sync/clean.sh
selecting podman as container runtime
Cleaning up ...
error: the server doesn't have a resource type "ot"
error: the server doesn't have a resource type "dv"
error: the server doesn't have a resource type "dataimportcron"
error: the server doesn't have a resource type "cdi"
No resources found
No resources found
No resources found ```

This seems to have nothing to do with the bazel-build-builder.sh script

**Comment by awels** [comment]
The error appears to be
```
 Error in download_and_extract: java.io.IOException: Error downloading [https://mirror.bazel.build/openjdk/azul-zulu11.50.19-ca-jdk11.0.12/zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz] to /root/.cache/bazel/_bazel_root/e03bf9038fc5089c0dbc8615812d9838/external/remotejdk11_linux/temp18280869944155108110/zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz: Bytes read 177470144 but wanted 202599005
ERROR: /root/.cache/bazel/_bazel_root/e03bf9038fc5089c0dbc8615812d9838/external/bazel_tools/tools/jdk/BUILD:29:19: @bazel_tools//tools/jdk:current_java_runtime depends on @remotejdk11_linux//:jdk in repository @remotejdk11_linux which failed to fetch. no such package '@remotejdk11_linux//': java.io.IOException: Error downloading [https://mirror.bazel.build/openjdk/azul-zulu11.50.19-ca-jdk11.0.12/zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz] to /root/.cache/bazel/_bazel_root/e03bf9038fc5089c0dbc8615812d9838/external/remotejdk11_linux/temp18280869944155108110/zulu11.50.19-ca-jdk11.0.12-linux_x64.tar.gz: Bytes read 177470144 but wanted 202599005
ERROR: Analysis of target '//cmd/cdi-cloner:go_default_test' failed; build aborted: 
```

Maybe a broken mirror?

**Comment by cfilleke** [comment]
It's got that look about it, doesn't it? Who would we alert to check on this, Alex? 
@awels

**Comment by awels** [comment]
/test pull-containerized-data-importer-e2e-upg
will try one more time, if not I can override since all the other lanes passed.

**Pull request merged**